core/src/main/kotlin/ch/ergon/dope/resolvable/clause/model/UseKeysClause.kt

```diff
@@ -0,0 +1,44 @@
+package ch.ergon.dope.resolvable.clause.model
+
+import ch.ergon.dope.DopeQuery
+import ch.ergon.dope.resolvable.clause.IDeleteClause
+import ch.ergon.dope.resolvable.clause.IDeleteUseKeysClause
+import ch.ergon.dope.resolvable.clause.ISelectFromClause
+import ch.ergon.dope.resolvable.clause.ISelectUseKeysClause
+import ch.ergon.dope.resolvable.expression.TypeExpression
+import ch.ergon.dope.validtype.ArrayType
+import ch.ergon.dope.validtype.StringType
+
+private const val USE_KEYS = "USE KEYS"
+
+sealed class SelectUseKeysClause(
+    private val parentDopeQuery: DopeQuery,
+    private val keysDopeQuery: DopeQuery,
+) : ISelectUseKeysClause {
+    override fun toDopeQuery() = DopeQuery(
+        queryString = "${parentDopeQuery.queryString} $USE_KEYS ${keysDopeQuery.queryString}",
+        parameters = parentDopeQuery.parameters + keysDopeQuery.parameters,
+    )
+}
+
+class SelectUseKeysStringClause(key: TypeExpression<StringType>, parentClause: ISelectFromClause) :
+    SelectUseKeysClause(parentClause.toDopeQuery(), key.toDopeQuery())
```

Comment:
Don't call the `.toDopeQuery` function, because of the parameters. The parameters should only be created, when calling the build function but with this they would be created, when creating an instance of this class.

Reply:
done, now the call is happening in the `toDopeQuery` function of their respective clauses